edx_event_bus_kafka/internal/tests/test_producer.py

```diff
@@ -137,3 +139,58 @@ def test_send_to_event_bus(self, mock_get_serializers):
             on_delivery=ep.on_event_deliver,
             headers={'ce_type': 'org.openedx.learning.auth.session.login.completed.v1'},
         )
+
+    @override_settings(EVENT_BUS_KAFKA_POLL_INTERVAL_SEC=0.05)
+    def test_polling_loop_terminates(self):
+        """
+        Test that polling loop stops as soon as the producer is garbage-collected.
+        """
+        call_count = 0
+
+        def increment_call_count(*args):
+            nonlocal call_count
+            call_count += 1
+
+        mock_producer = Mock(**{'poll.side_effect': increment_call_count})
+        producer_api = ep.EventProducerKafka(mock_producer)  # Created, starts polling
+
+        # Allow a little time to pass and check that the mock poll has been called
+        time.sleep(1.0)
+        assert call_count >= 3  # some small value; would actually be about 20
+        print(producer_api)  # Use the value here to ensure it isn't GC'd early
+
+        # Allow garbage collection of these objects, then ask for it to happen.
+        producer_api = None
+        mock_producer = None
+        gc.collect()
+
+        # Allow things to settle down post-GC (and let thread
+        # notice the object is gone), then save off the new call
+        # count.
+        time.sleep(0.2)
```

Comment:
A comment here about how you determined 0.2 would be useful (even if it was just heuristic).